nsqdserver/protocol_v2.go

```diff
@@ -1426,10 +1484,11 @@ func (p *protocolV2) internalPubAndTrace(client *nsqd.ClientV2, params [][]byte,
 		topic.GetDetailStats().UpdatePubClientStats(client.String(), client.UserAgent, "tcp", 1, false)
 		cost := time.Now().UnixNano() - startPub
 		topic.GetDetailStats().UpdateTopicMsgStats(int64(len(realBody)), cost/1000)
-		if !traceEnable {
-			return okBytes, nil
+
+		if existInJsonHeader || (traceEnable && !existInJsonHeader) {
```

Comment:
Maybe add var `needTracedRsp` to stand for whether we need return with `getTracedResponse`